libs/core/kiln_ai/adapters/ml_model_list.py

```diff
@@ -700,6 +702,49 @@ class KilnModel(BaseModel):
             ),
         ],
     ),
+    # Claude Sonnet 4
+    KilnModel(
+        family=ModelFamily.claude,
+        name=ModelName.claude_sonnet_4,
+        friendly_name="Claude Sonnet 4",
+        providers=[
+            KilnModelProvider(
+                name=ModelProviderName.openrouter,
+                model_id="anthropic/claude-sonnet-4",
+                suggested_for_data_gen=True,
+                suggested_for_evals=True,
+            ),
+            KilnModelProvider(
+                name=ModelProviderName.anthropic,
+                model_id="claude-sonnet-4-20250514",
+                structured_output_mode=StructuredOutputMode.json_instructions,
```

Comment:
Why not `structured_output_mode=StructuredOutputMode.function_calling` like 3.7?

JSON instructions inject extra copy into the prompt to get JSON, and don't have any structure guarantees. Generally the preference order is json_schema, function_calling, json_object, json_object_and_instruction, json_instruction. The earlier ones are more formal systems of structured data, the later are "ask and hope".